pynamodb/attributes.py

```diff
@@ -334,6 +357,52 @@ def __ne__(self, other: Any) -> bool:
         return self is not other
 
 
+class DiscriminatorAttribute(Attribute[type]):
+    attr_type = 'S'
+
+    def __init__(self, attr_name: Optional[str] = None):
+        super().__init__(attr_name=attr_name)
+        self.class_map: Dict[type, Any] = {}
+        self.discriminator_map: Dict[Any, type] = {}
+
+    def register_class(self, cls: type, discriminator: Any):
+        discriminator = discriminator(cls) if callable(discriminator) else discriminator
+        current_class = self.discriminator_map.get(discriminator)
+        if current_class and current_class != cls:
+            raise ValueError("The discriminator value '{}' is already assigned to a class: {}".format(
+                discriminator, current_class.__name__))
+
+        current_discriminator = self.class_map.get(cls)
+        if current_discriminator is not None:
+            self.discriminator_map.pop(current_discriminator)
```

Comment:
`pop` discarding result is `del`. Or did you want `pop(current_discriminator, None)`?
```suggestion
            del self.discriminator_map[current_discriminator]
```